src/util/antoraSupport.ts

```diff
@@ -0,0 +1,36 @@
+import vscode from 'vscode'
+import fs from 'fs'
+import yaml from 'js-yaml'
+import path from 'path'
+
+export default class AntoraSupport {
+  public async isDetected (): Promise<boolean> {
+    const antoraYmlFiles = await vscode.workspace.findFiles('**/antora.yml', '/node_modules/', 10)
+    return antoraYmlFiles.length > 0
+  }
+
+  public async findModulesDirPath (textDocument: vscode.TextDocument): Promise<any> {
+    const pathToAsciidocFile = textDocument.uri.fsPath
+    const antoraYmlFiles = await vscode.workspace.findFiles('**/antora.yml', '/node_modules/', 100)
+    // check if antora.yml file matches with textDocument content source root
+    try {
+      for (let i = 0; i < antoraYmlFiles.length; i++) {
+        if (pathToAsciidocFile.startsWith(path.dirname(antoraYmlFiles[i].path))) {
+          return pathToAsciidocFile.substring(0, pathToAsciidocFile.indexOf('modules'))
+        }
+      }
+    } catch (err) {
+      await vscode.window.showErrorMessage('Antora modules is out of the current workspace, ' + err.toString())
```

Comment:
I don't think we should show an error message. If we cannot find any `antora.yml` that's fine.
Using `console.log` is probably enough.